include/seqan3/io/sequence/sequence_file_format_sam.hpp

```diff
@@ -0,0 +1,352 @@
+            write_qual(stream_it, options, seq_qual);
+        }
+        else{
+          stream_it = '*';
+        }
+        stream_it = '\n';
+    }
+protected:
+    //!\privatesection
+    //!\brief Implementation of reading the ID.
+    template <typename stream_view_t,
+              typename seq_legal_alph_type,
+              typename id_type>
+    void read_id(stream_view_t                                       & stream_view,
+                 sequence_file_in_options<seq_legal_alph_type> const & options,
+                 id_type                                             & id)
+    {
+        // read id until first tab
+        ranges::copy(stream_view | ranges::view::take_while(!(is_char<'\t'>{}) ),
+                          detail::make_conversion_output_iterator(id));
+    }
+    //!\brief Implementation of reading the sequence.
+    template <typename      stream_view_t,
+              typename      seq_legal_alph_type,
+              range_concept seq_type>
+    void read_seq(stream_view_t                                       & stream_view,
+                  sequence_file_in_options<seq_legal_alph_type>  const & options,
+                  seq_type                                            & seq)
+    {
+        is_in_alphabet<seq_legal_alph_type> const is_legal_alph;
+        auto const is_tab = is_char<'\t'>{};
+        // read seq by jumping about over all the infos in sam file we are not interested in
+        ranges::copy(stream_view | ranges::view::drop_while(is_tab)
+                                 | ranges::view::drop_while(!is_tab)
+                                 | ranges::view::drop_while(is_tab)
+                                 | ranges::view::drop_while(!is_tab)
+                                 | ranges::view::drop_while(is_tab)
+                                 | ranges::view::drop_while(!is_tab)
+                                 | ranges::view::drop_while(is_tab)
+                                 | ranges::view::drop_while(!is_tab)
+                                 | ranges::view::drop_while(is_tab)
+                                 | ranges::view::drop_while(!is_tab)
+                                 | ranges::view::drop_while(is_tab)
+                                 | ranges::view::drop_while(!is_tab)
+                                 | ranges::view::drop_while(is_tab)
+                                 | ranges::view::drop_while(!is_tab)
+                                 | ranges::view::drop_while(is_tab)
+                                 | ranges::view::drop_while(!is_tab)
+                                 | ranges::view::drop_while(is_tab)
+                                 | ranges::view::take_while(!is_tab)                // until next
+                                 | ranges::view::drop_while(is_tab)
+                                 | ranges::view::remove_if(is_space || is_digit)   // ignore whitespace and numbers
+                                 | view::transform([is_legal_alph] (char const c)
```

Comment:
Does this work? It shouldn't! The view doesn't actually start doing anything until you start iterating over it...

Instead do multiple `detail::consume(stream_view | ranges::view::take_while(is_tab);` beforehand

Reply:
> Does this work? It shouldn't! 

Actually it should work, but it will be slower than doing the consumes I mentioned above.